### PR TITLE
Update docs and add an upgrade guidance doc

### PR DESCRIPTION
**Pull request by xinlili-statsig**

### CODE REVIEW

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+title: Node SDK v5.x.x to v6.0.0 Upgrade Guide
+---
+### Summary 
+
+Node.js SDK V6.0.0 introduced breaking changes on 
+- Removed async version of core APIs (checkGate, getConfig..)
+- Changed DataAdapter cache keys SDK used to read and write from
+- Changed default hashing algorithm used for getClientInitializeResponse() which is being used for bootstrapping client sdk.
+- If no environment tier is being set for SDK, we will default environment to be `production` when evaluating. 
+- Changed how you should disable exposure logging
```
Comment (tore-statsig) [inline docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:12]: each of these should link to a section below that explains it in more detail.  So 

`#### Synchronous Core APIs` is a good start, but you need one for each line

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+slug: /server/nodejsServerSDK/upgrade_guide/v5ToV6
+title: Node SDK v5.x.x to v6.0.0 Upgrade Guide
+---
+### Summary 
+
+Node.js SDK V6.0.0 introduced breaking changes on 
+- Removed async version of core APIs (checkGate, getConfig..)
+- Changed DataAdapter cache keys SDK used to read and write from
+- Changed default hashing algorithm used for getClientInitializeResponse() which is being used for bootstrapping client sdk.
+- If no environment tier is being set for SDK, we will default environment to be `production` when evaluating.
```
Comment (tore-statsig) [inline docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:11]: not entirely true, is it?  Dont we have the option to configure this now?

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+**Migration Guide**
+If you are using gate as following way, and you are using **javascript runtime error** will be reported, please audit and update the usage
+If you are using typescript, compile error will be reported.
+Meanwhile, we deprecated functions `*Sync()` for example `checkGateSync()`, please use checkGate() 
+```javascript
+Statsig.checkGate(user, "gate").then(() => {
+  // do somethig
+}).catch() // Compile error if you are using typescript. If you are using javascript, there will be runtime error please audit and 
+```
+If you re using as following ways, still recommend to remove await, but there won't be any error reported.
```
Comment (tore-statsig) [inline docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:47]: have you verified this is true?

**Comment by jasonwzm** [comment]
🚀 Deployed on https://6735265989dd226d285e899a--cozy-fox-0defec.netlify.app

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+const gate = await Statsig.checkGate(user, "gate")
+```
+#### DataAdapter Cache Key Changes
+**Context**
+To better support using multiple Statsig Instances in one runtime, and also better support over different responses format, for example download_config_spec_v1 vs download_config_spec_v2, we updated the cache keys SDK read and write from to avoid cache key collision. 
+
+**Changes**
+- Change cache key for config_specs from `statsig.cache` to 
+
+ `statsig|v1/download_config_specs|{String(useGzip)}|{SHA256Hashed(secretkey)}`
```
Comment (tore-statsig) [inline docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:58]: worth noting they should make it able to handle v2/download_config_specs as that may come soon?

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+#### Hash Algorithm Change on getClientInitializeResponse() 
+**Context**
+In order to reduce payload size, we changed default hash algorithm being used for all config names to be djb2 instead of SHA256. 
+
+**Changes**
+Config (gate / experiment / layers) names are now hashed with djb2 algorithm instead of SHA256 by default. If you are specifying what hash algorithm to use, this should not have affect on you. 
+
+**Migration Guide**
+If you are using latest (or relatively new client sdk), client side should automatically supported, no actions needed from you.
```
Comment (tore-statsig) [inline docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:82]: we should pull these versions and list out all client sdks

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+**Migration Guide**
+1. Update your DataAdapter if you are customizing the key and get and read logic 
+2. If you are writing to cache in other places (instead of relying on SDK setting the cache), update the call-site to use the new key
+3. If you are ok with cache miss with the first request, then you don’t need to do anything, SDK will write to cache with new key.
+4. If you don’t want cold cache, please write with the new key before push to production
+
+#### Hash Algorithm Change on getClientInitializeResponse() 
+**Context**
+In order to reduce payload size, we changed default hash algorithm being used for all config names to be djb2 instead of SHA256.
```
Comment (tore-statsig) [inline docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:76]: its also faster for evaluation

File: docs/server/node/_getConfig.mdx [hunk 7]
```diff
@@ -1,5 +1,5 @@
 ```jsx
-const config = Statsig.getConfigSync(user, "awesome_product_details");
```
Comment (tore-statsig) [inline docs/server/node/_getConfig.mdx:2]: we need something at the TOP of the node sdk about this.  Nothing shows until the bottom

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx [hunk 8]
```diff
@@ -0,0 +1,108 @@
+---
```
Comment (tore-statsig) [inline docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:1]: this page has no sidebar

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx (hunk 5 above)
Comment (xinlili-statsig) [reply docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:82]: Make sense. I am thinking about a separate page?

File: docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx (hunk 8 above)
Comment (xinlili-statsig) [reply docs/server/node/upgrade_guide/v5ToV6UpgradeGuide.mdx:1]: Should we have a sidebar for this page? I deliberately not have a sidebar for this page, it's only linked within the page

**Pull request merged**